### PR TITLE
Feature/external share

### PR DESCRIPTION
**Pull request by Shahar-Y**
Added the external share component. Tested with docker-compose.
A running version of the app can be found at:
http://test-sh.northeurope.cloudapp.azure.com/files
(with the 'merges' branch of api-gw)

### CODE REVIEW

File: public/index.html
Comment (yonatandt) [inline public/index.html]: delete the spinner

File: src/api/index.js
Comment (yonatandt) [inline src/api/index.js:16]: add`exShare`  to the export object

File: src/api/users.js
Comment (yonatandt) [inline src/api/users.js:19]: open an issue in `delegation-service` so it will return the user in the correct format

File: src/api/users.js
Comment (yonatandt) [inline src/api/users.js]: should not be in this file

File: src/components/Sidebar.vue
Comment (yonatandt) [inline src/components/Sidebar.vue]: delete this function

File: src/components/Sidebar.vue
Comment (yonatandt) [inline src/components/Sidebar.vue]: remove `onMySharesClick`

File: src/utils/constants.js
Comment (yonatandt) [inline src/utils/constants.js:37]: Verify this list with Kidon again before merging

File: src/components/buttons/ChangeShare.vue
Comment (yonatandt) [inline src/components/buttons/ChangeShare.vue]: Where is this file used?

File: src/components/common/EditApproversList.vue
Comment (yonatandt) [inline src/components/common/EditApproversList.vue:2]: what is the difference between this and `edit-permission-list`?

File: src/components/common/PermissionList.vue
Comment (yonatandt) [inline src/components/common/PermissionList.vue]: delete comments

File: src/components/common/PermissionList.vue
Comment (yonatandt) [inline src/components/common/PermissionList.vue]: this function has no use

File: src/components/common/PermitList.vue
Comment (yonatandt) [inline src/components/common/PermitList.vue]: delete comment

File: src/components/common/PermitList.vue
Comment (yonatandt) [inline src/components/common/PermitList.vue:1]: should it be the same component as `permission-list`?

File: src/components/files/AutoSuggestor.vue
Comment (yonatandt) [inline src/components/files/AutoSuggestor.vue]: bad name. Anyway, the class is not used.

File: src/components/files/StepOne.vue
Comment (yonatandt) [inline src/components/files/StepOne.vue]: bad name. Anyway, the id is not used.

File: src/components/files/StepOne.vue
Comment (yonatandt) [inline src/components/files/StepOne.vue]: Why not use Array includes()?

File: src/components/files/StepTwo.vue
Comment (yonatandt) [inline src/components/files/StepTwo.vue]: class not used

File: src/components/files/StepTwo.vue
Comment (yonatandt) [inline src/components/files/StepTwo.vue:61]: who is using the return value of this function?

File: src/components/files/StepTwo.vue
Comment (yonatandt) [inline src/components/files/StepTwo.vue]: Translate the message.

File: src/css/tabs.css
Comment (yonatandt) [inline src/css/tabs.css]: delete comments

File: src/css/tabs.css
Comment (yonatandt) [inline src/css/tabs.css]: delete comment

File: src/css/tabs.css
Comment (yonatandt) [inline src/css/tabs.css]: delete comment

File: src/components/prompts/ShareExternal.vue
Comment (yonatandt) [inline src/components/prompts/ShareExternal.vue]: bad name for the function

File: src/components/prompts/Share.vue
Comment (yonatandt) [inline src/components/prompts/Share.vue:1]: We need to refactor this file and shorten it drastically.

**Review by yonatandt — CHANGES_REQUESTED** [review]

File: src/components/common/EditApproversList.vue
Comment (Shahar-Y) [reply src/components/common/EditApproversList.vue:2]: The functionality is different, mainly for saving the approvers list at the end of the external share, but also in the future I would want them to look different, since these are different operations.

File: src/components/common/PermitList.vue
Comment (Shahar-Y) [reply src/components/common/PermitList.vue:1]: It is possible to make them the same with some modifications, but you created both of them so you should be the one to decide

File: src/components/files/AutoSuggestor.vue
Comment (Shahar-Y) [reply src/components/files/AutoSuggestor.vue]: :(

File: src/components/files/StepOne.vue
Comment (Shahar-Y) [reply src/components/files/StepOne.vue]: I am officially insulted.

File: src/components/files/StepOne.vue
Comment (Shahar-Y) [reply src/components/files/StepOne.vue]: 'includes' won't work for an array of objects, but I made it look better now

File: src/components/prompts/ShareExternal.vue
Comment (Shahar-Y) [reply src/components/prompts/ShareExternal.vue]: Not sure about that. what would you suggest?

File: src/components/prompts/Share.vue
Comment (Shahar-Y) [reply src/components/prompts/Share.vue:1]: Agreed

File: public/index.html
Comment (Shahar-Y) [reply public/index.html]: done

File: src/api/index.js
Comment (Shahar-Y) [reply src/api/index.js:16]: done

File: src/api/users.js
Comment (Shahar-Y) [reply src/api/users.js:19]: done

File: src/api/users.js
Comment (Shahar-Y) [reply src/api/users.js]: moved to utils

File: src/components/Sidebar.vue
Comment (Shahar-Y) [reply src/components/Sidebar.vue]: done

File: src/components/Sidebar.vue
Comment (Shahar-Y) [reply src/components/Sidebar.vue]: done

File: src/utils/constants.js
Comment (Shahar-Y) [reply src/utils/constants.js:37]: I took it from Kidon

File: src/components/buttons/ChangeShare.vue
Comment (Shahar-Y) [reply src/components/buttons/ChangeShare.vue]: Deleted

File: src/components/common/PermissionList.vue
Comment (Shahar-Y) [reply src/components/common/PermissionList.vue]: done

File: src/components/common/PermissionList.vue
Comment (Shahar-Y) [reply src/components/common/PermissionList.vue]: deleted

File: src/components/common/PermitList.vue
Comment (Shahar-Y) [reply src/components/common/PermitList.vue]: done

File: src/components/files/AutoSuggestor.vue
Comment (Shahar-Y) [reply src/components/files/AutoSuggestor.vue]: deleted

File: src/components/files/StepOne.vue
Comment (Shahar-Y) [reply src/components/files/StepOne.vue]: deleted

File: src/components/files/StepTwo.vue
Comment (Shahar-Y) [reply src/components/files/StepTwo.vue]: removed

File: src/components/files/StepTwo.vue
Comment (Shahar-Y) [reply src/components/files/StepTwo.vue:61]: good catch. deleted

File: src/components/files/StepTwo.vue
Comment (Shahar-Y) [reply src/components/files/StepTwo.vue]: done

File: src/css/tabs.css
Comment (Shahar-Y) [reply src/css/tabs.css]: done

File: src/css/tabs.css
Comment (Shahar-Y) [reply src/css/tabs.css]: done

File: src/css/tabs.css
Comment (Shahar-Y) [reply src/css/tabs.css]: done

File: src/components/common/PermitList.vue
Comment (yonatandt) [reply src/components/common/PermitList.vue:1]: Opened new issue #139

File: src/components/prompts/ShareExternal.vue
Comment (yonatandt) [reply src/components/prompts/ShareExternal.vue]: A function name which starts with `is`, usually returns a Boolean value, which this function does not.
I would suggest `activateStep` by looking at line 10. But I'm still not completely what's the function's purpose.

File: src/components/prompts/Share.vue
Comment (yonatandt) [reply src/components/prompts/Share.vue:1]: Did you open an issue for it?

Comment (Shahar-Y) [reply src/components/prompts/Share.vue:1]: No. Working on it in feature/beautify-share

**Comment by Shahar-Y** [comment]
As for the security vulnerability, I talked to one of Snyk's security analysts who referred me to this issue - https://github.com/npm/cli/issues/782 in npm itself. The vulnerability is known to npm and I was told that currently the best option is to either ignore it, or fork from every repo on the dependency chain and fix the dependency - which seems to be too much effort in this case.
Therefore, I suggest to ignore it for now.

File: src/components/prompts/ShareExternal.vue
Comment (Shahar-Y) [reply src/components/prompts/ShareExternal.vue]: You are right. fixed it in #140

File: src/components/prompts/Share.vue
Comment (Shahar-Y) [reply src/components/prompts/Share.vue:1]: #140

**Comment by Shahar-Y** [comment]
Updated the kubernetes in http://test-sh.northeurope.cloudapp.azure.com/files

**Pull request merged**
